### PR TITLE
Feature new report

### PR DESCRIPTION
**Pull request by weigandjakob**

### CODE REVIEW

File: test/test.js [hunk 1]
```diff
@@ -1,35 +1,461 @@
...
+                        },
+                        counter: {
+                            type: "number",
+                            writable: true,
+                            observable: true,
+                            readOnly: false,
+                            writeOnly: false,
+                            forms: [
+                                {
+                                    href:
```
Comment (egekorkan) [inline test/test.js:173]: would be better to not split lines here?

File: test/test.js [hunk 2]
```diff
@@ -1,35 +1,461 @@
...
+                                        "coap://localhost:8082/TestServient/pr/counter",
+                                    contentType: "application/json",
+                                    op: ["readproperty", "writeproperty"],
+                                },
+                            ],
+                        },
+                        temperature: {
+                            type: "number",
+                            writable: false,
```
Comment (egekorkan) [inline test/test.js:195]: writable could be removed

File: test/test.js [hunk 3]
```diff
@@ -1,35 +1,461 @@
...
+                                {
+                                    href:
+                                        "coap://localhost:8082/TestServient/ev/onChange",
+                                    contentType: "application/json",
+                                    op: "subscribeevent",
+                                },
+                            ],
+                        },
+                    },
+                    id: "urn:uuid:4d856a78-a2ff-4dff-90c2-af42afce13e0",
```
Comment (egekorkan) [inline test/test.js:431]: must make sure that this matches the produced TD. If id is not given explicitly, node-wot creates one

Comment (weigandjakob) [reply test/test.js:431]: Then id: `"urn:uuid:4d856a78-a2ff-4dff-90c2-af42afce13e0",` should probably be removed? PerfectThing has no id statement. Other possible solutions would be to ad an id statement to perfectThing. Which solution should i choose?

Comment (weigandjakob) [reply test/test.js]: Added id statement to perfectThing.js TD.

File: test/test.js (hunk 1 above)
Comment (weigandjakob) [reply test/test.js:168]: Was done by prettier automatically.

File: test/test.js (hunk 2 above)
Comment (weigandjakob) [reply test/test.js]: writable statements are removed in newest commit.

File: src/TestReport.ts [hunk 4]
```diff
@@ -1,14 +1,133 @@
...
+        super(testCycle, testScenario, name);
+        this.report = new MiniTestReport();
+    }
+
+    /**
+     * Restructures the ReportContainer to match the structure of the printed Message. Potentially a new JSON object
+     * could be created and returned here if in the future the originally formatted container is needed for further processing.
+     * @return The restructured testReportContainer.
+     */
+    getPrintableMessage() {
```
Comment (egekorkan) [inline src/TestReport.ts:65]: Why actually deleting them but not printing what is needed?

File: test/test.js [hunk 5]
```diff
@@ -1,35 +1,457 @@
...
+}
+
+/**
+ * Returns all TestCase JSON objects with the given name from the given Array of TestCase JSON object.
+ * @param {[JSON]} testCases The array of JSON objects containing the TestCases to check.
+ * @param {string} testCaseName The string containing the TestCase name.
+ * @returns An arry containing all the TestCase JSON objects with the given name. Array is empty
+ * if no TestCase with the given name has been found.
+ */
+function getAllTestCasesWithName(testCases, testCaseName) {
```
Comment (egekorkan) [inline test/test.js:103]: might be better to call byName to be similar to the function above

File: src/TestReport.ts (hunk 4 above)
Comment (weigandjakob) [reply src/TestReport.ts:65]: The original container is not needed any further and this way a new object does not have to be created &rarr; less code is needed which reduces maintenance and improves readability. But i can create a new JSON object here if you want.

File: test/test.js (hunk 5 above)
Comment (weigandjakob) [reply test/test.js:103]: I thought getAllTestCasesWithName makes more sence than getAllTestCasesByName. Would Also further differentiate the methods that return a single testCase from the ones that return multiple if in the future more methods like this are introduced. But can change it if you want.

**Pull request merged**